src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas1/Cas1SpaceSearchService.kt

```diff
@@ -19,11 +24,23 @@ import java.util.UUID
 class Cas1SpaceSearchService(
   private val characteristicService: CharacteristicService,
   private val spaceSearchRepository: Cas1SpaceSearchRepository,
+  private val applicationRepository: ApplicationRepository,
 ) {
   fun findSpaces(searchParameters: Cas1SpaceSearchParameters): List<Cas1SpaceSearchResult> {
+    val applicationId = searchParameters.applicationId
+    val application = applicationRepository.findByIdOrNull(searchParameters.applicationId)
+      ?: throw NotFoundProblem(applicationId, "Application")
+    if (application !is ApprovedPremisesApplicationEntity) {
```

Comment:
Optional - If we added an `ApprovedPremiseApplicationRepository` we wouldn't need to do the type check. We've done something similar for `ApprovedPremisesRepository`